target_parquet/__init__.py

```diff
@@ -19,23 +19,24 @@
 from enum import Enum
 from multiprocessing import Process, Queue
 
-from .helpers import flatten, flatten_schema
+from .helpers import flatten, flatten_schema, flatten_schema_to_pyarrow_schema
 
 _all__ = ["main"]
 
 LOGGER = singer.get_logger()
 LOGGER.setLevel(os.getenv("LOGGER_LEVEL", "INFO"))
 
 
-def create_dataframe(list_dict, remove_empty_columns=False):
+def create_dataframe(list_dict, schema, force_output_schema_cast=False):
     fields = set()
     for d in list_dict:
         fields = fields.union(d.keys())
-    dataframe = pa.table({f: [row.get(f) for row in list_dict] for f in fields
-                          if any([row.get(f) for row in list_dict]) or not remove_empty_columns})
+    data = {f: [row.get(f) for row in list_dict] for f in fields}
+    dataframe = pa.table(data)
+    if force_output_schema_cast and schema:

```

Comment:
raise an exception if `force_output_schema_cast=True` but `schema` is not provided.